reliability/core-it/src/androidTest/kotlin/com/datadog/android/core/integration/FeatureSdkCoreContextTest.kt

```diff
@@ -0,0 +1,638 @@
+        val fakeEvent1 = forge.anAlphabeticalString()
+        val fakeEvent2 = forge.anAlphabeticalString()
+        testedFeatureSdkCore.registerFeature(stubFeature)
+        testedFeatureSdkCore.setEventReceiver(stubFeature.name, stubFeatureEventReceiver)
+        val stubRegisteredFeature = testedFeatureSdkCore.getFeature(stubFeature.name)
+        checkNotNull(stubRegisteredFeature)
+        stubRegisteredFeature.sendEvent(fakeEvent1)
+
+        // When
+        testedFeatureSdkCore.removeEventReceiver(stubFeature.name)
+        stubRegisteredFeature.sendEvent(fakeEvent2)
+
+        // Then
+        assertThat(stubFeatureEventReceiver.getReceivedEvents()).hasSize(1)
+        assertThat(stubFeatureEventReceiver.getReceivedEvents()).containsOnly(fakeEvent1)
+    }
+
+    @Test
+    fun mustNotReceiveEvents_when_eventReceiverRegistered_eventSentForADifferentFeature() {
+        // Given
+        val stubFeatureEventReceiver = StubFeatureEventReceiver()
+        val otherFeature = StubStorageBackedFeature(
+            forge,
+            forge.anAlphabeticalString(),
+            getMockServerWrapper().getServerUrl()
+        )
+        val fakeEvent = forge.anAlphabeticalString()
+        testedFeatureSdkCore.registerFeature(stubFeature)
+        testedFeatureSdkCore.registerFeature(otherFeature)
+        testedFeatureSdkCore.setEventReceiver(stubFeature.name, stubFeatureEventReceiver)
+        val otherRegisteredFeature = testedFeatureSdkCore.getFeature(otherFeature.name)
+
+        // When
+        checkNotNull(otherRegisteredFeature)
+        otherRegisteredFeature.sendEvent(fakeEvent)
+
+        // Then
+        assertThat(stubFeatureEventReceiver.getReceivedEvents()).isEmpty()
+    }
+
+    @Test
+    fun mustReceiveEvents_when_eventReceiverRegistered_sendEventCalledFromDifferentThreads() {
+        // Given
+        val fakeEvent1 = forge.anAlphabeticalString()
+        val fakeEvent2 = forge.anAlphabeticalString()
+        val countDownLatch = CountDownLatch(2)
+        val stubFeatureEventReceiver = StubFeatureEventReceiver(countDownLatch::countDown)
+        testedFeatureSdkCore.registerFeature(stubFeature)
+        testedFeatureSdkCore.setEventReceiver(stubFeature.name, stubFeatureEventReceiver)
+        val stubRegisteredFeature = testedFeatureSdkCore.getFeature(stubFeature.name)
+
+        // When
+        checkNotNull(stubRegisteredFeature)
+        Thread {
+            stubRegisteredFeature.sendEvent(fakeEvent1)
+        }.start()
+        Thread {
+            stubRegisteredFeature.sendEvent(fakeEvent2)
+        }.start()
+
+        // Then
+        countDownLatch.await(SHORT_WAIT_MS, TimeUnit.MILLISECONDS)
+        assertThat(stubFeatureEventReceiver.getReceivedEvents()).containsExactlyInAnyOrder(fakeEvent1, fakeEvent2)
+    }
+
+    // endregion
+
+    // region SingleThreadExecutor Service
+
+    @Test
+    fun mustCreateASingleThreadExecutorService_when_createSingleThreadExecutorService() {
```

Comment:
executors created in this test and in the next one will be running until test suite finishes the execution, taking some memory. Should we shut them down in the end of the test?

Reply:
yup good call